apps/ledger-live-mobile/src/families/evm/EditTransactionFlow/MethodSelection.tsx

```diff
@@ -0,0 +1,240 @@
+  }
+
+  return "editTransaction.speedUp.description";
+};
+
+type Props = StackNavigatorProps<
+  EditTransactionParamList,
+  ScreenName.EditTransactionMethodSelection
+>;
+
+function MethodSelectionComponent({ navigation, route }: Props) {
+  const { t } = useTranslation();
+  const { operation, account, parentAccount } = route.params;
+
+  invariant(
+    operation.transactionRaw,
+    "operation.transactionRaw not found. Could not edit the transaction",
+  );
+
+  const [transactionHasBeenValidated, setTransactionHasBeenValidated] = useState(false);
+
+  const mainAccount = getMainAccount(account, parentAccount);
+
+  const transactionToEdit = fromTransactionRaw(
+    operation.transactionRaw as TransactionRaw,
+  ) as EvmTransaction;
+
+  const { transaction, setTransaction } = useBridgeTransaction<EvmTransaction>(() => ({
+    account,
+    parentAccount,
+    transaction: transactionToEdit,
+  }));
+
+  invariant(
+    transaction,
+    "[useBridgeTransaction - MethodSelection] could not found transaction from bridge.",
+  );
+
+  const haveFundToCancel = hasMinimumFundsToCancel({
+    mainAccount,
+    transactionToUpdate: transactionToEdit,
+  });
+
+  const haveFundToSpeedup = hasMinimumFundsToSpeedUp({
+    account,
+    mainAccount,
+    transactionToUpdate: transactionToEdit,
+  });
+
+  const [selectedMethod, setSelectedMethod] = useState<EditType | null>();
+
+  // if we are at this step (i.e: in this screen) it means the transaction is editable
+  const isOldestEditableOperation = isOldestPendingOperation(mainAccount, transactionToEdit.nonce);
+  const bridge: AccountBridge<EvmTransaction> = getAccountBridge(account, parentAccount as Account);
+
+  const onSelect = useCallback(
+    async (option: EditType) => {
+      log("Edit Transaction - Method Selection", "onSelect Cancel/Speed up", option);
+
+      const patch = await getEditTransactionPatch({
+        account: mainAccount,
+        transaction: transactionToEdit,
+        editType: option,
+      });
+
+      setTransaction(bridge.updateTransaction(transaction, patch));
+
+      setSelectedMethod(option);
+    },
+    [mainAccount, transaction, transactionToEdit, bridge, setTransaction],
+  );
+
+  /**
+   * Poll for transaction confirmation
+   * If the transaction has been confirmed, we navigate to the error screen
+   * PS: this still runs even if the user has navigated forward in the steps
+   * This might be a ReactNative antipatern leading to silent error
+   * "Cannot update a component while rendering a different component" error
+   * But it works for now and avoid code duplication 🤷‍♂️
+   * If you have a better way to do this, please do!
+   */
+  useEffect(() => {
+    const setTransactionHasBeenValidatedCallback = async () => {
+      const tx = await getTransactionByHash(mainAccount.currency, operation.hash);
+      if (tx?.confirmations) {
+        // stop polling as soon as we have a confirmation
+        clearInterval(intervalId);
```

Comment:
shouldn't we call this function after intervalId is defined ?

Reply:
clearInterval never throws, no matter the param you give it.